client/src/main/java/io/getstream/chat/android/client/helpers/AttachmentHelper.kt

```diff
@@ -0,0 +1,21 @@
+package io.getstream.chat.android.client.helpers
+
+import io.getstream.chat.android.client.models.Attachment
+import io.getstream.chat.android.client.utils.SystemTimeProvider
+import okhttp3.HttpUrl.Companion.toHttpUrlOrNull
+
+class AttachmentHelper(private val systemTimeProvider: SystemTimeProvider = SystemTimeProvider()) {
+
+    fun hasValidUrl(attachment: Attachment): Boolean {
+        val url = attachment.url?.toHttpUrlOrNull() ?: return false
+        if (url.queryParameterNames.contains(QUERY_KEY_NAME_EXPIRES).not()) {
```

Comment:
Could also be expressed like this, which you might find more readable:

```suggestion
        if (QUERY_KEY_NAME_EXPIRES !in url.queryParameterNames) {
```

Reply:
Don't see big difference here. Just matter of taste. Thanks anyway